Gems/Terrain/Code/Source/TerrainRenderer/TerrainMeshManager.cpp

```diff
@@ -523,52 +520,90 @@ namespace Terrain
         }
     }
 
-    void TerrainMeshManager::InitializeTerrainPatch(uint16_t gridSize, PatchData& patchdata)
+    void TerrainMeshManager::InitializeTerrainPatch(PatchData& patchdata)
     {
         patchdata.m_xyPositions.clear();
         patchdata.m_indices.clear();
 
-        const uint16_t gridVertices = gridSize + 1; // For m_gridSize quads, (m_gridSize + 1) vertices are needed.
-        const size_t size = gridVertices * gridVertices;
-
-        patchdata.m_xyPositions.reserve(size);
+        // Moser–de Bruijn sequences for x and y
+        AZStd::array<uint16_t, GridVerts1D> zOrderX;
+        AZStd::array<uint16_t, GridVerts1D> zOrderY;
```

Comment:
Suggestion: This would be a good place to explain what's going on a little bit more, and also provide a bit more definition around "z-order" which has a double meaning - it can either mean "order in the z coordinate", or "produce a z-order curve". I was reading it as the first meaning, which wasn't making sense to me.

Maybe something like this:
```
/*
 Generate the terrain patch using Moser-de Bruijn sequences for x and y. These sequences create 
a Z-order (Morton) curve, which produces a more cache-friendly ordering of triangles vertices than 
a linear ordering of vertices by increasing the probability that neighboring vertices will be in the cache.
*/
```

Reply:
I worked on the commenting some and added a larger comment at the top of the function explaining what the function does as a whole to give subsequent comments more context.